File: QuestionBankApi/Services/ExamService.cs
using Microsoft.EntityFrameworkCore;
using QuestionBankApi.Data;
using QuestionBankApi.DTOs;
using QuestionBankApi.Enums;
using QuestionBankApi.Models;
using QuestionBankApi.LoginSystemApi.Data;

namespace QuestionBankApi.Services;

/// <summary>
/// 试卷服务实现
/// </summary>
public class ExamService : IExamService
{
    private readonly QuestionBankDbContext _db;
    private readonly UserDBContext _userDb;
    private readonly IQuestionService _questionService;

    /// <summary>
    /// 构造函数
    /// </summary>
    /// <param name="db">数据库上下文</param>
    /// <param name="userDb">用户数据库上下文</param>
    /// <param name="questionService">题目服务</param>
    public ExamService(QuestionBankDbContext db, UserDBContext userDb, IQuestionService questionService)
    {
        _db = db;
        _userDb = userDb;
        _questionService = questionService;
    }

    /// <summary>
    /// 创建试卷
    /// </summary>
    /// <param name="dto">试卷数据传输对象</param>
    /// <returns>创建的试卷</returns>
    public async Task<ExamDto> CreateExamAsync(ExamDto dto)
    {
        // 创建试卷实体
        var exam = new Exam
        {
            Title = dto.Title,
            Description = dto.Description,
            CreatedAt = DateTime.Now,
            Deadline = dto.Deadline,
            Status = dto.Status,
            TotalScore = dto.Questions.Sum(q => q.Score)
        };

        // 添加试卷
        _db.Exams.Add(exam);
        await _db.SaveChangesAsync();

        // 添加试卷题目
        foreach (var questionDto in dto.Questions)
        {
            var examQuestion = new ExamQuestion
            {
                ExamId = exam.Id,
                QuestionId = questionDto.QuestionId,
                Order = questionDto.Order,
                Score = questionDto.Score
            };
            _db.ExamQuestions.Add(examQuestion);
        }

        await _db.SaveChangesAsync();

        // 返回创建的试卷
        var createdExam = await GetExamByIdAsync(exam.Id);
        return createdExam ?? throw new Exception($"无法获取新创建的试卷，ID: {exam.Id}");
    }

    /// <summary>
    /// 获取所有试卷
    /// </summary>
    /// <returns>试卷列表</returns>
    public async Task<IEnumerable<ExamDto>> GetAllExamsAsync()
    {
        var exams = await _db.Exams
            .Include(e => e.ExamQuestions)
            .OrderByDescending(e => e.CreatedAt)
            .ToListAsync();

        return exams.Select(e => new ExamDto
        {
            Id = e.Id,
            Title = e.Title,
            Description = e.Description,
            CreatedAt = e.CreatedAt,
            Deadline = e.Deadline,
            TotalScore = e.TotalScore,
            Status = e.Status,
            Questions = e.ExamQuestions.Select(q => new ExamQuestionDto
            {
                QuestionId = q.QuestionId,
                Order = q.Order,
                Score = q.Score
            }).ToList()
        });
    }

    /// <summary>
    /// 根据ID获取试卷
    /// </summary>
    /// <param name="id">试卷ID</param>
    /// <returns>试卷信息</returns>
    public async Task<ExamDto?> GetExamByIdAsync(int id)
    {
        var exam = await _db.Exams
            .Include(e => e.ExamQuestions)
            .ThenInclude(q => q.Question)
            .FirstOrDefaultAsync(e => e.Id == id);

        if (exam == null)
            return null;

        var examDto = new ExamDto
        {
            Id = exam.Id,
            Title = exam.Title,
            Description = exam.Description,
            CreatedAt = exam.CreatedAt,
            Deadline = exam.Deadline,
            TotalScore = exam.TotalScore,
            Status = exam.Status,
            Questions = new List<ExamQuestionDto>()
        };

        foreach (var examQuestion in exam.ExamQuestions.OrderBy(q => q.Order))
        {
            var question = examQuestion.Question;
            var questionDto = question != null ? new QuestionDto
            {
                Id = question.Id,
                Content = question.Content,
                Type = question.Type,
                OptionsJson = question.OptionsJson,
                AnswersJson = question.AnswersJson,
                Analysis = question.Analysis,
                Difficulty = question.Difficulty,
                TagsJson = question.TagsJson,
                CreateTime = question.CreateTime.ToString("yyyy-MM-dd HH:mm:ss")
            } : null;

            examDto.Questions.Add(new ExamQuestionDto
            {
                QuestionId = examQuestion.QuestionId,
                Order = examQuestion.Order,
                Score = examQuestion.Score,
                Question = questionDto
            });
        }

        return examDto;
    }

    /// <summary>
    /// 获取学生的试卷列表
    /// </summary>
    /// <param name="studentId">学生ID</param>
    /// <returns>试卷列表</returns>
    public async Task<IEnumerable<ExamDto>> GetExamsByStudentIdAsync(int studentId)
    {
        var assignments = await _db.ExamAssignments
            .Include(a => a.Exam)
            .ThenInclude(e => e != null ? e.ExamQuestions : null)
            .Where(a => a.StudentId == studentId)
            .ToListAsync();

        // 过滤掉没有试卷关联的分配记录
        var validAssignments = assignments.Where(a => a.Exam != null).ToList();

        return validAssignments.Select(a => new ExamDto
        {
            Id = a.Exam!.Id,
            Title = a.Exam.Title,
            Description = a.Exam.Description,
            CreatedAt = a.Exam.CreatedAt,
            Deadline = a.Exam.Deadline,
            TotalScore = a.Exam.TotalScore,
            Status = a.Exam.Status,
            Questions = a.Exam.ExamQuestions.Select(q => new ExamQuestionDto
            {
                QuestionId = q.QuestionId,
                Order = q.Order,
                Score = q.Score
            }).ToList()
        });
    }

    /// <summary>
    /// 分配试卷给学生
    /// </summary>
    /// <param name="examId">试卷ID</param>
    /// <param name="studentIds">学生ID列表</param>
    /// <returns>分配结果</returns>
    public async Task<bool> AssignExamToStudentsAsync(int examId, List<int> studentIds)
    {
        // 检查试卷是否存在
        var exam = await _db.Exams.FindAsync(examId);
        if (exam == null)
            return false;

        // 检查学生是否存在
        var existingStudents = await _userDb.Users
            .Where(u => studentIds.Contains(u.Id) && u.Role == LoginSystemApi.Models.UserRole.Student)
            .Select(u => u.Id)
            .ToListAsync();

        if (existingStudents.Count != studentIds.Count)
            return false;

        // 检查是否已分配
        var existingAssignments = await _db.ExamAssignments
            .Where(a => a.ExamId == examId && studentIds.Contains(a.StudentId))
            .Select(a => a.StudentId)
            .ToListAsync();

        // 添加新分配
        var newAssignments = studentIds
            .Except(existingAssignments)
            .Select(studentId => new ExamAssignment
            {
                ExamId = examId,
                StudentId = studentId,
                AssignedAt = DateTime.Now
            });

        _db.ExamAssignments.AddRange(newAssignments);

        // 更新试卷状态为已发布
        if (exam.Status == 0)
        {
            exam.Status = 1; // 已发布
        }

        await _db.SaveChangesAsync();
        return true;
    }

    /// <summary>
    /// 检查试卷是否分配给学生
    /// </summary>
    /// <param name="examId">试卷ID</param>
    /// <param name="studentId">学生ID</param>
    /// <returns>是否分配</returns>
    public async Task<bool> IsExamAssignedToStudentAsync(int examId, int studentId)
    {
        return await _db.ExamAssignments
            .AnyAsync(a => a.ExamId == examId && a.StudentId == studentId);
    }

    /// <summary>
    /// 提交试卷
    /// </summary>
    /// <param name="examId">试卷ID</param>
    /// <param name="studentId">学生ID</param>
    /// <param name="dto">提交数据</param>
    /// <returns>提交结果</returns>
    public async Task<ExamResultDto> SubmitExamAsync(int examId, int studentId, ExamSubmissionDto dto)
    {
        // 检查试卷是否存在
        var exam = await _db.Exams
            .Include(e => e.ExamQuestions)
            .FirstOrDefaultAsync(e => e.Id == examId);

        if (exam == null)
            throw new Exception($"未找到ID为{examId}的试卷");

        // 检查是否已分配给学生
        var assignment = await _db.ExamAssignments
            .FirstOrDefaultAsync(a => a.ExamId == examId && a.StudentId == studentId);

        if (assignment == null)
            throw new Exception("该试卷未分配给当前学生");

        // 检查是否已提交
        if (assignment.IsSubmitted)
            throw new Exception("该试卷已提交，不能重复提交");

        // 检查截止时间
        if (exam.Deadline.HasValue && DateTime.Now > exam.Deadline.Value)
            throw new Exception("已超过截止时间，不能提交");

        // 创建提交记录
        var submission = new ExamSubmission
        {
            ExamId = examId,
            StudentId = studentId,
            SubmittedAt = DateTime.Now,
            CompletionTime = dto.CompletionTime,
            Score = 0 // 初始分数为0，后面计算
        };

        _db.ExamSubmissions.Add(submission);
        await _db.SaveChangesAsync();

        // 计算得分
        int totalScore = 0;
        int correctCount = 0;

        foreach (var answerDto in dto.Answers)
        {
            // 获取题目信息
            var question = await _db.Questions.FindAsync(answerDto.QuestionId);
            if (question == null)
                continue;

            // 获取题目在试卷中的分值
            var examQuestion = exam.ExamQuestions.FirstOrDefault(q => q.QuestionId == answerDto.QuestionId);
            if (examQuestion == null)
                continue;

            // 判断答案是否正确
            bool isCorrect = IsAnswerCorrect((int)question.Type, question.AnswersJson, answerDto.Answer);
            int score = isCorrect ? examQuestion.Score : 0;

            // 创建答案记录
            var answer = new QuestionAnswer
            {
                SubmissionId = submission.Id,
                QuestionId = answerDto.QuestionId,
                Answer = answerDto.Answer,
                IsCorrect = isCorrect,
                Score = score
            };

            _db.QuestionAnswers.Add(answer);

            // 累计得分
            totalScore += score;
            if (isCorrect)
                correctCount++;
        }

        // 更新提交记录的得分
        submission.Score = totalScore;

        // 更新分配记录为已提交
        assignment.IsSubmitted = true;

        await _db.SaveChangesAsync();

        // 返回提交结果
        var student = await _userDb.Users.FindAsync(studentId);

        return new ExamResultDto
        {
            ExamId = examId,
            StudentId = studentId,
            StudentName = student?.Username,
            SubmittedAt = submission.SubmittedAt,
            CompletionTime = submission.CompletionTime,
            TotalScore = exam.TotalScore,
            Score = totalScore,
            CorrectCount = correctCount,
            QuestionCount = exam.ExamQuestions.Count,
            Answers = dto.Answers.Select(a => new QuestionAnswerDto
            {
                QuestionId = a.QuestionId,
                Answer = a.Answer,
                IsCorrect = _db.QuestionAnswers
                    .FirstOrDefault(qa => qa.SubmissionId == submission.Id && qa.QuestionId == a.QuestionId)?.IsCorrect ?? false,
                Score = _db.QuestionAnswers
                    .FirstOrDefault(qa => qa.SubmissionId == submission.Id && qa.QuestionId == a.QuestionId)?.Score ?? 0
            }).ToList()
        };
    }

    /// <summary>
    /// 获取试卷统计信息
    /// </summary>
    /// <param name="examId">试卷ID</param>
    /// <returns>统计信息</returns>
    public async Task<ExamStatisticsDto> GetExamStatisticsAsync(int examId)
    {
        // 检查试卷是否存在
        var exam = await _db.Exams
            .Include(e => e.ExamQuestions)
            .ThenInclude(q => q.Question)
            .FirstOrDefaultAsync(e => e.Id == examId);

        if (exam == null)
            throw new Exception($"未找到ID为{examId}的试卷");

        // 获取所有提交记录
        var submissions = await _db.ExamSubmissions
            .Include(s => s.Answers)
            .Where(s => s.ExamId == examId)
            .ToListAsync();

        // 获取所有分配记录
        var assignments = await _db.ExamAssignments
            .Where(a => a.ExamId == examId)
            .ToListAsync();

        // 计算统计信息
        var statistics = new ExamStatisticsDto
        {
            ExamId = examId,
            ExamTitle = exam.Title,
            StudentCount = assignments.Count,
            SubmittedCount = submissions.Count,
            AverageScore = submissions.Any() ? submissions.Average(s => s.Score) : 0,
            HighestScore = submissions.Any() ? submissions.Max(s => s.Score) : 0,
            LowestScore = submissions.Any() ? submissions.Min(s => s.Score) : 0,
            PassRate = submissions.Any() ? (double)submissions.Count(s => s.Score >= exam.TotalScore * 0.6) / submissions.Count : 0,
            ScoreDistribution = new Dictionary<string, int>
            {
                { "0-59", submissions.Count(s => s.Score < exam.TotalScore * 0.6) },
                { "60-69", submissions.Count(s => s.Score >= exam.TotalScore * 0.6 && s.Score < exam.TotalScore * 0.7) },
                { "70-79", submissions.Count(s => s.Score >= exam.TotalScore * 0.7 && s.Score < exam.TotalScore * 0.8) },
                { "80-89", submissions.Count(s => s.Score >= exam.TotalScore * 0.8 && s.Score < exam.TotalScore * 0.9) },
                { "90-100", submissions.Count(s => s.Score >= exam.TotalScore * 0.9) }
            },
            QuestionStatistics = new List<QuestionStatisticsDto>(),
            StudentResults = new List<ExamResultDto>()
        };

        // 计算各题统计信息
        foreach (var examQuestion in exam.ExamQuestions.OrderBy(q => q.Order))
        {
            var question = examQuestion.Question;
            if (question == null)
                continue;

            var answers = submissions
                .SelectMany(s => s.Answers)
                .Where(a => a.QuestionId == question.Id)
                .ToList();

            var correctCount = answers.Count(a => a.IsCorrect);
            var incorrectCount = answers.Count - correctCount;

            var questionStatistics = new QuestionStatisticsDto
            {
                QuestionId = question.Id,
                Order = examQuestion.Order,
                Type = (int)question.Type,
                Content = question.Content,
                CorrectCount = correctCount,
                IncorrectCount = incorrectCount,
                CorrectRate = answers.Any() ? (double)correctCount / answers.Count : 0
            };

            // 如果是选择题，计算各选项的选择人数
            if (question.Type == QuestionType.Single || question.Type == QuestionType.Judge)
            {
                var optionCounts = new Dictionary<string, int>();
                foreach (var answer in answers)
                {
                    var options = answer.Answer.Split(',');
                    foreach (var option in options)
                    {
                        if (!string.IsNullOrEmpty(option))
                        {
                            if (optionCounts.ContainsKey(option))
                                optionCounts[option]++;
                            else
                                optionCounts[option] = 1;
                        }
                    }
                }
                questionStatistics.OptionCounts = optionCounts;
            }

            statistics.QuestionStatistics.Add(questionStatistics);
        }

        // 获取学生成绩
        var results = await GetExamResultsAsync(examId);
        statistics.StudentResults = results.ToList();

        return statistics;
    }

    /// <summary>
    /// 获取试卷成绩
    /// </summary>
    /// <param name="examId">试卷ID</param>
    /// <param name="studentId">学生ID（可选）</param>
    /// <returns>成绩信息</returns>
    public async Task<IEnumerable<ExamResultDto>> GetExamResultsAsync(int examId, int? studentId = null)
    {
        // 构建查询
        var query = _db.ExamSubmissions
            .Include(s => s.Answers)
            .Where(s => s.ExamId == examId);

        // 如果指定了学生ID，只查询该学生的成绩
        if (studentId.HasValue)
        {
            query = query.Where(s => s.StudentId == studentId.Value);
        }

        // 执行查询
        var submissions = await query.ToListAsync();

        // 获取试卷信息
        var exam = await _db.Exams
            .Include(e => e.ExamQuestions)
            .FirstOrDefaultAsync(e => e.Id == examId);

        if (exam == null)
            throw new Exception($"未找到ID为{examId}的试卷");

        // 获取学生信息
        var studentIds = submissions.Select(s => s.StudentId).Distinct().ToList();
        var students = await _userDb.Users
            .Where(u => studentIds.Contains(u.Id))
            .ToDictionaryAsync(u => u.Id, u => u.Username);

        // 构建结果
        var results = new List<ExamResultDto>();
        foreach (var submission in submissions)
        {
            var result = new ExamResultDto
            {
                ExamId = examId,
                StudentId = submission.StudentId,
                StudentName = students.ContainsKey(submission.StudentId) ? students[submission.StudentId] : null,
                SubmittedAt = submission.SubmittedAt,
                CompletionTime = submission.CompletionTime,
                TotalScore = exam.TotalScore,
                Score = submission.Score,
                CorrectCount = submission.Answers.Count(a => a.IsCorrect),
                QuestionCount = exam.ExamQuestions.Count,
                Answers = submission.Answers.Select(a => new QuestionAnswerDto
                {
                    QuestionId = a.QuestionId,
                    Answer = a.Answer,
                    IsCorrect = a.IsCorrect,
                    Score = a.Score
                }).ToList()
            };

            results.Add(result);
        }

        return results;
    }

    /// <summary>
    /// 判断答案是否正确
    /// </summary>
    /// <param name="questionType">题目类型</param>
    /// <param name="correctAnswer">正确答案</param>
    /// <param name="studentAnswer">学生答案</param>
    /// <returns>是否正确</returns>
    private static bool IsAnswerCorrect(int questionType, string? correctAnswer, string studentAnswer)
    {
        // 去除空格
        correctAnswer = correctAnswer?.Trim() ?? "";
        studentAnswer = studentAnswer?.Trim() ?? "";

        // 单选题和判断题
        if (questionType == 1 || questionType == 3)
        {
            return string.Equals(correctAnswer, studentAnswer, StringComparison.OrdinalIgnoreCase);
        }
        // 多选题
        else if (questionType == 2)
        {
            // 分割答案
            var correctOptions = correctAnswer.Split(',').Select(o => o.Trim()).Where(o => !string.IsNullOrEmpty(o)).OrderBy(o => o).ToArray();
            var studentOptions = studentAnswer.Split(',').Select(o => o.Trim()).Where(o => !string.IsNullOrEmpty(o)).OrderBy(o => o).ToArray();

            // 比较
            return correctOptions.SequenceEqual(studentOptions, StringComparer.OrdinalIgnoreCase);
        }
        // 填空题和简答题
        else if (questionType == 4 || questionType == 5)
        {
            // 简单比较，实际应用中可能需要更复杂的比较逻辑
            return string.Equals(correctAnswer, studentAnswer, StringComparison.OrdinalIgnoreCase);
        }

        return false;
    }
}

File: src/services/examService.ts
import { api } from './api';
// 指向新的 exam 类型文件
import type { ExamStatisticsDto, ExamResultDto } from '../types/exam';

// 响应接口定义
interface ApiResponse<T> {
  success: boolean;
  data: T;
  message?: string;
}

interface ExamResponse {
  id: number;
  title: string;
  description?: string;
  questions: any[];
  totalScore?: number;
  deadline?: string;
  isSubmitted?: boolean;
}

interface ExamSubmissionResponse {
  data: {
    score: number;
    totalScore: number;
    correctCount: number;
    questionCount: number;
    completionTime: number;
    submittedAt: string;
    answers: Array<{
      questionId: number;
      answer: string;
      isCorrect: boolean;
      score: number;
    }>;
  };
}

// 试卷服务
export const examService = {
  // 获取所有试卷（管理员）
  async getExams(): Promise<ApiResponse<ExamResponse[]>> {
    try {
      const response = await api.get<ApiResponse<ExamResponse[]>>('/exams');
      console.log('获取试卷原始响应:', response);

      // 如果返回空数组，创建一个空数组作为默认值
      if (!response) {
        console.warn('试卷响应为空');
        return { success: true, data: [] };
      }

      // 处理不同的响应格式
      if (response.success && Array.isArray(response.data)) {
        // 标准格式，直接返回
        return response;
      } else if (Array.isArray(response)) {
        // 如果响应本身就是数组，将其包装为标准格式
        return { success: true, data: response };
      } else if (response.data === null || response.data === undefined) {
        // 如果 data 字段为空，返回空数组
        console.warn('试卷响应的 data 字段为空');
        return { success: true, data: [] };
      }

      // 默认情况，直接返回原始响应
      return response as ApiResponse<ExamResponse[]>;
    } catch (error) {
      console.error('获取试卷列表失败:', error);
      // 返回空数组作为默认值，避免前端报错
      return { success: false, data: [] };
    }
  },

  // 获取学生的试卷
  async getStudentExams(): Promise<ExamResponse[]> {
    try {
      const response = await api.get<ExamResponse[]>('/exams');
      return Array.isArray(response) ? response : [];
    } catch (error) {
      console.error('获取学生试卷列表失败:', error);
      throw error;
    }
  },

  // 获取试卷详情
  async getExamById(id: number): Promise<ExamResponse> {
    try {
      return await api.get<ExamResponse>(`/exams/${id}`);
    } catch (error) {
      console.error(`获取试卷 ${id} 详情失败:`, error);
      throw error;
    }
  },

  // 创建试卷
  async createExam(examData: any) {
    try {
      const response = await api.post('/exams', examData);
      console.log('创建试卷响应:', response);
      return response;
    } catch (error) {
      console.error('创建试卷失败:', error);
      throw error;
    }
  },

  // 更新试卷
  async updateExam(examData: any) {
    try {
      return await api.put(`/exams/${examData.id}`, examData);
    } catch (error) {
      console.error(`更新试卷 ${examData.id} 失败:`, error);
      throw error;
    }
  },

  // 删除试卷
  async deleteExam(id: number) {
    try {
      return await api.delete(`/exams/${id}`);
    } catch (error) {
      console.error(`删除试卷 ${id} 失败:`, error);
      throw error;
    }
  },

  // 获取学生列表
  async getStudents(): Promise<ApiResponse<any[]>> {
    try {
      console.log('获取学生列表...');
      const response = await api.get<ApiResponse<any[]>>('/users/students');
      console.log('获取学生列表响应:', response);

      // 处理不同的响应格式
      if (response && response.success && Array.isArray(response.data)) {
        // 标准格式，直接返回
        return response;
      } else if (Array.isArray(response)) {
        // 如果响应本身就是数组，将其包装为标准格式
        return { success: true, data: response };
      } else if (!response || !response.data) {
        // 如果没有数据，返回空数组
        console.warn('学生列表为空');
        return { success: true, data: [] };
      }

      // 默认情况，直接返回原始响应
      return response as ApiResponse<any[]>;
    } catch (error) {
      console.error('获取学生列表失败:', error);
      throw error; // 抛出错误，让调用者处理
    }
  },

  // 分配试卷给学生
  async assignExam(examId: number, studentIds: number[]) {
    try {
      console.log(`准备分配试卷 ${examId} 给学生:`, studentIds);
      const response = await api.post(`/exams/${examId}/assign`, { studentIds });
      console.log(`分配试卷 ${examId} 响应:`, response);
      return response;
    } catch (error) {
      console.error(`分配试卷 ${examId} 失败:`, error);
      throw error; // 抛出错误，让调用者处理
    }
  },

  // 提交试卷
  async submitExam(examId: number, submission: any): Promise<ExamSubmissionResponse> {
    try {
      console.log(`正在提交试卷 ${examId}，数据:`, JSON.stringify(submission).substring(0, 200) + '...');
      
      // 添加超时处理
      const timeoutPromise = new Promise<never>((_, reject) => {
        setTimeout(() => reject(new Error('提交请求超时，请重试')), 20000); // 延长到20秒
      });

      // 实际请求
      const fetchPromise = api.post<ExamSubmissionResponse>(`/exams/${examId}/submit`, submission);

      // 使用Promise.race竞争，哪个先完成就返回哪个
      try {
        const response = await Promise.race([fetchPromise, timeoutPromise]);
        console.log(`试卷 ${examId} 提交成功，响应:`, response);
        return response;
      } catch (raceError: any) {
        console.error(`提交试卷 Promise.race 出错:`, raceError);
        throw raceError;
      }
    } catch (error: any) {
      console.error(`提交试卷 ${examId} 失败:`, error);
      
      // 针对特定错误提供更友好的消息
      if (error.message && error.message.includes('无法连接到服务器')) {
        throw new Error(`服务器连接失败，请确认后端服务已启动并且API地址正确 (${error.message})`);
      }
      
      throw error;
    }
  },

  /**
   * 获取试卷统计信息 (管理员)
   * @param examId 试卷ID
   * @returns 试卷统计数据
   */
  async getExamStatistics(examId: number): Promise<ExamStatisticsDto> {
    try {
      console.log(`准备获取试卷 ${examId} 的统计信息`);
      // 后端返回的数据结构是 { success: true, data: statistics }
      const response = await api.get<{ success: boolean; data: ExamStatisticsDto; message?: string }>(`/exams/${examId}/statistics`);
      
      if (response && response.success && response.data) {
         console.log(`获取试卷 ${examId} 统计信息响应:`, response.data);
         return response.data; // 直接返回 data 部分
      } else {
        // 如果 success 为 false 或 data 不存在，则抛出错误
        throw new Error(response?.message || `获取统计数据失败 (${examId})`);
      }
    } catch (error) {
      console.error(`获取试卷 ${examId} 统计信息失败:`, error);
      // 可以考虑向上层抛出更具体的错误，或者保留原始错误
      throw error; 
    }
  },

  // 获取试卷结果
  async getExamResults(examId: number, studentId?: number) {
    try {
      const url = studentId
        ? `/exams/${examId}/results?studentId=${studentId}`
        : `/exams/${examId}/results`;
      return await api.get(url);
    } catch (error) {
      console.error(`获取试卷 ${examId} 结果失败:`, error);
      throw error;
    }
  },

  // 导出试卷结果
  async exportExamResults(examId: number) {
    try {
      return await api.get(`/exams/${examId}/export`, {
        responseType: 'blob'
      });
    } catch (error) {
      console.error(`导出试卷 ${examId} 结果失败:`, error);
      throw error;
    }
  }
};
